hack/README.md

```diff
@@ -0,0 +1,17 @@
+# hack GuideLines
+
+This document describes how you can use the scripts from [`hack`](.) directory
+and gives a bried introduction and explanation of these scripts.
+
+## Overview
+
+The [`hack`](.) directory contains many scripts that ensure continuous development of cnf features,
+enhance the robustness of the code, improve developmetn efficiency, etc.
+The explanations and descriptions of these sciprts are helpful for contributors.
+
+## setup-test-cluster.sh
+This script does all the CNF installation parts that are expected to be done by users, beside installation related ones. like labeling nodes and creating needed MachineConfiguration object, etc.
```

Comment:
This is a bit confusing IMHO: "it does all the installation parts beside installation".
What about: 
This script does all the cluster setup parts that are expected to be done by admins, like labelling nodes and creating the MachineConfigPool resource, etc.

Reply:
I wanted to convey that it does not install CNF just do the pre/post setup
still not sure this is good enough

Reply:
maybe use your phrasing and add after the dot. "It does not install CNF itself"